server/game/core/ability/abilityTargets/PlayerTargetResolver.ts

```diff
@@ -0,0 +1,83 @@
+import { TargetResolverBaseClass } from './TargetResolverBaseClass';
+import { IPlayerTargetResolver } from '../../../TargetInterfaces';
+import { AbilityContext } from '../AbilityContext';
+import type Player from '../../Player';
+import PlayerOrCardAbility from '../PlayerOrCardAbility';
+import { Stage } from '../../Constants';
+
+export class PlayerTargetResolver extends TargetResolverBaseClass<IPlayerTargetResolver<AbilityContext>> {
+    public constructor(name: string, properties: IPlayerTargetResolver<AbilityContext>, ability: PlayerOrCardAbility) {
+        super(name, properties, ability);
+
+        if (this.properties.immediateEffect) {
+            this.properties.immediateEffect.setDefaultTargetFn((context) => context.targets[name]);
+        }
+    }
+
+    protected override hasLegalTarget(context: any): boolean {
+        // I don't think there are cases of player-targeting abilities that are sometimes invalid, but I could be wrong.
+        return true;
+    }
+
+    protected override hasTargetsChosenByInitiatingPlayer(context: AbilityContext): boolean {
+        return this.getChoosingPlayer(context) === context.player;
+    }
+
+    protected override checkTarget(context: AbilityContext): boolean {
+        return context.targets[this.name] && context.game.getPlayers().includes(context.targets[this.name]);
+    }
+
+    protected override getAllLegalTargets(context: AbilityContext): Player[] {
+        return context.game.getPlayers();
+    }
+
+    protected override resolve(context: AbilityContext, targetResults, passPrompt = null) {
+        if (targetResults.cancelled || targetResults.payCostsFirst || targetResults.delayTargeting) {
+            return;
+        }
+
+        const player = context.choosingPlayerOverride || this.getChoosingPlayer(context);
+        if (player === context.player.opponent && context.stage === Stage.PreTarget) {
+            targetResults.delayTargeting = this;
+            return;
+        }
+        const promptTitle = this.properties.activePromptTitle || 'Choose a player';
+        const choices = ['You', 'Opponent'];
```

Comment:
Similar comment here - I think based on the way that the CardTargetResolver behaves, people would expect to have a little more control over this using filters / properties and more automation determining if a given player is a legal target or not. We need the behavior between the two to be as consistent as possible.

This is a probably large and definitely non-urgent task so feel free to freeze this part and just do the DrawSystem updates if you prefer. That said, definitely it would improve some cards in the long run. Up to you.